x-pack/qa/repository-old-versions-7x/src/javaRestTest/java/org/elasticsearch/oldrepos7x/OldRepositoryAccessIT.java

```diff
@@ -0,0 +1,525 @@
+        assertEquals(indexVersion.toReleaseVersion(), getResp.evaluate("snapshots.0.version"));
+
+        // list specific snapshot on new ES
+        getSnaps = new Request("GET", "/_snapshot/" + repoName + "/" + snapshotName);
+        getResponse = client().performRequest(getSnaps);
+        getResp = ObjectPath.createFromResponse(getResponse);
+        assertThat(getResp.evaluate("total"), equalTo(1));
+        assertThat(getResp.evaluate("snapshots.0.snapshot"), equalTo(snapshotName));
+        assertThat(getResp.evaluate("snapshots.0.repository"), equalTo(repoName));
+        assertThat(getResp.evaluate("snapshots.0.indices"), contains(indexName));
+        assertThat(getResp.evaluate("snapshots.0.state"), equalTo(SnapshotState.SUCCESS.toString()));
+        assertEquals(numberOfShards, (int) getResp.evaluate("snapshots.0.shards.successful"));
+        assertEquals(numberOfShards, (int) getResp.evaluate("snapshots.0.shards.total"));
+        assertEquals(0, (int) getResp.evaluate("snapshots.0.shards.failed"));
+        assertEquals(indexVersion.toReleaseVersion(), getResp.evaluate("snapshots.0.version"));
+
+        // list advanced snapshot info on new ES
+        getSnaps = new Request("GET", "/_snapshot/" + repoName + "/" + snapshotName + "/_status");
+        getResponse = client().performRequest(getSnaps);
+        getResp = ObjectPath.createFromResponse(getResponse);
+        assertThat(((List<?>) getResp.evaluate("snapshots")).size(), equalTo(1));
+        assertThat(getResp.evaluate("snapshots.0.snapshot"), equalTo(snapshotName));
+        assertThat(getResp.evaluate("snapshots.0.repository"), equalTo(repoName));
+        assertThat(((Map<?, ?>) getResp.evaluate("snapshots.0.indices")).keySet(), contains(indexName));
+        assertThat(getResp.evaluate("snapshots.0.state"), equalTo(SnapshotState.SUCCESS.toString()));
+        assertEquals(numberOfShards, (int) getResp.evaluate("snapshots.0.shards_stats.done"));
+        assertEquals(numberOfShards, (int) getResp.evaluate("snapshots.0.shards_stats.total"));
+        assertEquals(0, (int) getResp.evaluate("snapshots.0.shards_stats.failed"));
+        assertThat(getResp.evaluate("snapshots.0.stats.total.size_in_bytes"), greaterThan(0));
+        assertThat(getResp.evaluate("snapshots.0.stats.total.file_count"), greaterThan(0));
+
+        // restore / mount and check whether searches work
+        restoreMountAndVerify(numDocs, expectedIds, numberOfShards, sourceOnlyRepository, indexName, repoName, snapshotName);
+
+        // close indices
+        closeIndex(client(), "restored_" + indexName);
+        closeIndex(client(), "mounted_full_copy_" + indexName);
+        closeIndex(client(), "mounted_shared_cache_" + indexName);
+
+        // restore / mount again
+        restoreMountAndVerify(numDocs, expectedIds, numberOfShards, sourceOnlyRepository, indexName, repoName, snapshotName);
+
+        // TODO restart current cluster
```

Comment:
The original test this is copied from used to check restored indices are there [after a restart of the target cluster ](https://github.com/elastic/elasticsearch/blob/main/x-pack/qa/repository-old-versions/src/test/java/org/elasticsearch/oldrepos/OldRepositoryAccessIT.java#L126)(the one with the current version).
Orchestration of this was done through [special gradle tasks](https://github.com/elastic/elasticsearch/blob/main/x-pack/qa/repository-old-versions/build.gradle#L141) probably coming from the bwc-test plugin that we haven't introduced here to simplify things.

Doing this with the new testcluster framework should be possible by restarting the "currentCluster", the problem I can into when trying this though is that the existing client/adminClient from ESRestTestCase are loosing their connection and closing / restarting them wasn't as straight forward when I tried. Probably possible though with some more investigation.

Reply:
@javanna since we talked about this last I've added restarts to this PR.